File: clients/algoliasearch-client-php/lib/Model/Search/LogType.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * LogType Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class LogType
{
    /**
     * Possible values of this enum
     */
    const ALL = 'all';

    const QUERY = 'query';

    const BUILD = 'build';

    const ERROR = 'error';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::ALL,
            self::QUERY,
            self::BUILD,
            self::ERROR,
        ];
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/com/algolia/model/search/LogType.java
package com.algolia.model.search;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Gets or Sets logType */
@JsonAdapter(LogType.Adapter.class)
public enum LogType {
  ALL("all"),

  QUERY("query"),

  BUILD("build"),

  ERROR("error");

  private final String value;

  LogType(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static LogType fromValue(String value) {
    for (LogType b : LogType.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<LogType> {

    @Override
    public void write(final JsonWriter jsonWriter, final LogType enumeration)
      throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public LogType read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return LogType.fromValue(value);
    }
  }
}
